Kconfig.zephyr

```diff
@@ -625,118 +625,6 @@ config BOOTLOADER_SRAM_SIZE
 	  - Zephyr is a !XIP image, which implicitly assumes existence of a
 	  bootloader that loads the Zephyr !XIP image onto SRAM.
 
-config MCUBOOT
-	bool
-	help
-	  Hidden option used to indicate that the current image is MCUBoot
```

Comment:
It's a good idea to move this one (`CONFIG_MCUBOOT`), but the rest _could_ be used by an application without having the MCUboot module available?

Reply:
All but one of these options are used to configure the arguments passed to the `imgtool.py` script which resides in the MCUboot module. I would argue that these have no value if the MCUboot module is not available since the script is not available.

Then we are only left with `BOOTLOADER_MCUBOOT`. A user does not need this option to build an image that can be booted by MCUboot _without_ having the MCUboot module available. Instead the user can set the `ROM_START_OFFSET` to the appropriate value. Users that choose not to include MCUboot as a module, but instead choose to manually sign the images for use with MCUboot can be considered "expert users" IMO, and these could be expected to figure out that they need to set `ROM_START_OFFSET`.

Reply:
> All but one of these options are used to configure the arguments passed to the `imgtool.py` script which resides in the MCUboot module. I would argue that these have no value if the MCUboot module is not available since the script is not available.

Not true. The `imgtool.py` script can be installed in numerous ways - see https://github.com/zephyrproject-rtos/zephyr/blob/20cb60ecd25c7a1616c62de0ec07fb3eb4f4a037/scripts/west_commands/sign.py#L50 and https://github.com/zephyrproject-rtos/zephyr/blob/20cb60ecd25c7a1616c62de0ec07fb3eb4f4a037/scripts/west_commands/sign.py#L282